pkg/batchpertrace/batchpertrace.go

```diff
@@ -63,3 +63,51 @@ func Split(batch pdata.Traces) []pdata.Traces {
 
 	return result
 }
+
+// SplitLogs returns one pdata.Logs for each trace in the given pdata.Logs input. Each of the resulting pdata.Logs contains exactly one trace.
+func SplitLogs(batch pdata.Logs) []pdata.Logs {
+	// for each log in the resource logs, we group them into batches of rl/ill/traceID.
+	// if the same traceID exists in different ill, they land in different batches.
+	var result []pdata.Logs
+
+	for i := 0; i < batch.ResourceLogs().Len(); i++ {
+		rs := batch.ResourceLogs().At(i)
+
+		for j := 0; j < rs.InstrumentationLibraryLogs().Len(); j++ {
+			// the batches for this ILL
+			batches := map[[16]byte]pdata.ResourceLogs{}
+
+			ill := rs.InstrumentationLibraryLogs().At(j)
+			for k := 0; k < ill.Logs().Len(); k++ {
+				log := ill.Logs().At(k)
+				key := log.TraceID().Bytes()
```

Comment:
I would assume that not every log has a TraceID. Will it then be null?

Reply:
`.IsEmpty()` returns true for the TraceID, but TraceID itself is not null.